### PR TITLE
757 Add mini videos in the documentation

### PR DESCRIPTION
**Pull request by ipopescu**
### Related links

- [**New playlist**](https://www.youtube.com/playlist?list=PL8oWxbJ-csEqi5FP87EJZViE2aLz6X1Mj) created for writing on-chain code
- #757 

### Changes

We have created a new [**set of short videos**](https://www.youtube.com/playlist?list=PL8oWxbJ-csEqi5FP87EJZViE2aLz6X1Mj) to accompany the docs and to demonstrate the output of various interactions with the blockchain. 
This PR restructures the documentation and existing tutorials to include these mini videos in the writing-contracts section.
This PR includes the videos in the documentation since the Associtation finished them today (going further than the original task description).
The commits describe all the changes in more detail.

### Notes

The session code and contract code samples need to be revisited in a different card and PR; see #704.
I ran the site locally, and I checked the links and the videos.

### CODE REVIEW

File: source/docs/casper/dapp-dev-guide/writing-contracts/upgrading-contracts.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
 
 Our smart contract packaging tools enable you to:
 
--   [Upgrade your contracts](/dapp-dev-guide/tutorials/upgrade-contract.md) and specify how the state of the contract is managed
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/writing-contracts/upgrading-contracts.md:7]: I removed the link, so people don't jump off the page immediately. We have a video below.

**Comment by ipopescu** [comment]
@ACStoneCL this PR is ready for review. I moved most of the videos after the written explanation because I thought they would make more sense after some background is given. The videos are meant to demonstrate what the commands would output.

File: source/docs/casper/dapp-dev-guide/writing-contracts/getting-started.md [hunk 2]
```diff
@@ -2,15 +2,7 @@
...
-Casper's blockchain is built upon the Rust programming language and compiles down to WebAssembly. The Rust contract SDK is the easiest way to get started with smart contract development. This guide will walk you through the steps to set up your development environment and build your first contract.
-
-Refer to our [FAQ guide](faq/faq-developer.md) if you have questions or need help.
-
-## Video Tutorial {#video-tutorial}
-
-For a video walkthrough of this guide, feel free to check out this quick-start video.
-
-<iframe width="560" height="315" src="https://www.youtube.com/embed?v=XvV02iBoctc&list=PL8oWxbJ-csEogSV-M0IPiofWP5I_dLji6&index=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+Casper's blockchain is built upon the Rust programming language and compiles down to WebAssembly. The Rust contract SDK is the easiest way to get started with smart contract development. This guide will walk you through the steps to set up your development environment and build your first contract.
```
Comment (ACStoneCL) [inline source/docs/casper/dapp-dev-guide/writing-contracts/getting-started.md:5]: I don't disagree with this statement, but do we want to explicitly call out Rust as the easiest way?

**Review by ACStoneCL — APPROVED** [review]
Everything looks good to me, I just left the one comment about referencing Rust as the easiest way - I have no problem with it, just a question of how we want to go about it.

File: source/docs/casper/dapp-dev-guide/writing-contracts/getting-started.md (hunk 2 above)
Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/writing-contracts/getting-started.md:5]: Good catch. I don't think this sentence is adding value... we should remove it.
"Casper's blockchain is built upon the Rust programming language and compiles down to WebAssembly. This guide will walk you through the steps to set up your development environment and build your first contract."

**Comment by ipopescu** [comment]
@darthsiroftardis I will merge this content, just fyi.

**Pull request merged**